### ISSUE TITLE
Cmake build with GPU fails on Windows 10 with a MSB6006 error

### ISSUE THREAD

**Issue by Thimira** [tensorflow/tensorflow#19895]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 
- **TensorFlow installed from (source or binary)**: Source, from Git tag v1.8.0
- **TensorFlow version (use command below)**: 1.8 (Git tag v1.8.0)
- **Python version**: 3.5.5 (Anaconda Python)
- **Bazel version (if compiling from source)**: Cmake 3.11.3
- **GCC/Compiler version (if compiling from source)**: Visual Studio 2015 Update 3 (14.0.25431.01)
- **CUDA/cuDNN version**: CUDA 9.2, cuDNN 7.1
- **GPU model and memory**: Nvidia GeForce 940M
- **Exact command to reproduce**:
```
git clone https://github.com/tensorflow/tensorflow
....
git checkout v1.8.0
....
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\vcvarsall.bat amd64

cmake .. -A x64 -DCMAKE_BUILD_TYPE=Release ^
 -DSWIG_EXECUTABLE=C:/Dev_Tools/TFBuild/tools/swigwin-3.0.12/swig.exe ^
 -DPYTHON_EXECUTABLE=C:/Users/UAMARTH/AppData/Local/Continuum/anaconda3/envs/tfbuild/python.exe ^
 -DPYTHON_LIBRARIES=C:/Users/UAMARTH/AppData/Local/Continuum/anaconda3/envs/tfbuild/libs/python35.lib ^
 -Dtensorflow_ENABLE_GPU=ON ^
 -Dtensorflow_CUDA_VERSION=9.2 ^
 -Dtensorflow_CUDNN_VERSION=7 ^
 -DCUDNN_HOME="C:\Dev_Tools\TFBuild\tools\cudnn-9.2-windows10-x64-v7.1" ^
 -Dtensorflow_WIN_CPU_SIMD_OPTIONS=/arch:AVX


MSBuild /p:Configuration=Release tf_python_build_pip_package.vcxproj
```

### Describe the problem
The build fails with a "error MSB6006: "cmd.exe" exited with code 1". Full error message below.
The line 259 of referred to in the error message points to a "tensorflow\tensorflow\contrib\cmake\build\tf_python\tensorflow\tools\api\generator\api\keras\preprocessing\text\__init__.py" file, which does not exist. The .......\preprocessing\text\ directory is empty.

### Source code / logs
```
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (
1) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_api.vcxproj" (default target) (259) ->
(CustomBuild target) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" e
xited with code 1. [C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_api.vcxproj]

    19591 Warning(s)
    1 Error(s)

Time Elapsed 04:30:49.38
```

**Comment by angersson** [comment]
@gunan Can you take a look at this?

**Comment by gunan** [comment]
@annarev or @case540 may be able to help more on this failure. Looks like api generation failed?

**Comment by Thimira** [comment]
@annarev I have several anaconda environments setup on the machine I attempted to build TensorFlow, and one of the environments does have TensorFlow installed from pip. The anaconda environment which I attempted the build however does not have TensorFlow installed. My environment details are below,
```
conda info --env
# conda environments:
#
base                  *  C:\Users\UAMARTH\AppData\Local\Continuum\anaconda3
aws-tools                C:\Users\UAMARTH\AppData\Local\Continuum\anaconda3\envs\aws-tools
deep-learning            C:\Users\UAMARTH\AppData\Local\Continuum\anaconda3\envs\deep-learning
tfbuild                  C:\Users\UAMARTH\AppData\Local\Continuum\anaconda3\envs\tfbuild
```
From the above, the 'deep-learning' environment has TensorFlow installed from pip (tensorflow-gpu 1.5.0), while the other environments does not have it. I attempted the cmake build on the 'tfbuild' environment.

But, let me remove the environment with TensorFlow installed and re-run the build to see whether it has an effect.
Should I attempt the build with the current head commit from master branch, or should I go with a stable branch such as r1.9 or r1.8?

**Comment by annarev** [comment]
Sorry for not replying earlier. To answer your earlier question, any version is ok to try for building.
Do you see any other error in the log? The last one doesn't give much information. Can you try searching for "error :" or ": error" (don't remember which one is the right one)?

**Comment by Thimira** [comment]
@annarev I didn't see any other errors in the output. But let me re-run the build with 'detailed' verbosity and see if it shows any errors.

**Comment by Thimira** [comment]
@annarev I re-ran the build with detailed verbosity with the latest code from the master branch as of today (26/6/18 - 91780a3a338b09ef89f5f04ab8e683b9756b4bf0). The build still fails, but now it gives a different error.

```
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (
1) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\estimator_python_api.vcxproj" (default target) (2) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow_internal.vcxproj" (default target) (3
) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow_internal_static.vcxproj" (default tar
get) (4) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_stream_executor.vcxproj" (default target) (148) ->
(ClCompile target) ->
  C:\Dev_Tools\TFBuild\tensorflow\tensorflow/core/platform/test.h(34): fatal error C1083: Cannot open include file: 'gt
est/gtest.h': No such file or directory (compiling source file C:\Dev_Tools\TFBuild\tensorflow\tensorflow\stream_execut
or\lib\statusor_test.cc) [C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_stream_executor.vcxproj]

    25383 Warning(s)
    1 Error(s)

Time Elapsed 04:46:45.42
```

I did a search for "error:" in the build log as you suggested, which returned the following lines,

```
Testing ReaderTest/parseWithOneError: OK
Testing ReaderTest/parseChineseWithOneError: OK
Testing ReaderTest/parseWithDetailError: OK
```

": error" didn't have any matches.

Please find the complete build log here: [msbuild.log](https://www.dropbox.com/s/mu0axcrohet0zvp/msbuild.log?dl=0). Please note that the log file is about 100MB.

**Comment by gunan** [comment]
For the latest issue, I just pushed a fix to master. Could you try pulling head and retrying?

**Comment by Thimira** [comment]
@gunan I re-attempted the build, by cloning a fresh copy (with the head commit at bfcfad55b7b3fa4a1093fa748d4241f9457b2a84). However, it still failed with the following error,

```
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (
1) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\estimator_python_api.vcxproj" (default target) (2) ->
(CustomBuild target) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" e
xited with code 1. [C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\estimator_python_api.vcxproj]


"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (
1) ->
"C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_api.vcxproj" (default target) (264) ->
  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" e
xited with code 1. [C:\Dev_Tools\TFBuild\tensorflow\tensorflow\contrib\cmake\build\tf_python_api.vcxproj]

    25797 Warning(s)
    2 Error(s)

Time Elapsed 04:50:43.32
```

Please find the full build log [here](https://www.dropbox.com/s/w3id4nwic14q5uv/msbuild-1-7-2018.log?dl=0).

Question: does the build depend on any python modules already being installed? Can already installed python modules conflict with it? I have verified that tensorflow is not installed anywhere in my machine before building. But, I do have protobuf, and absl-py installed. Can it cause an issue?

**Comment by LoSealL** [comment]
For windows and cmake buiding, there is an issue when generating python api, in the scipt `create_python_api.py`. See my response to #20669 .

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gunan** [comment]
I think this was fixed, but I forgot to update this issue.
However, now this is obsolete because the recommended way to build TF on windows is now using bazel:
https://www.tensorflow.org/install/install_sources_windows

As a bonus, building TF on windows with GPU support is 5x faster using bazel than cmake!
